flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java

```diff
@@ -64,22 +64,37 @@
      *
      * <ul>
      *   <li>typeClass == atomic class (e.g. {@code Integer.class}) -> {@code ConfigOption<Integer>}
-     *   <li>typeClass == {@code Map.class} -> {@code ConfigOption<Map<String, String>>}
-     *   <li>typeClass == atomic class and isList == true for {@code ConfigOption<List<Integer>>}
+     *   <li>typeClass == atomic class and type == LIST for {@code ConfigOption<List<Integer>>}
```

Comment:
@mxm Please do not touch the config stack without a full JIRA ticket and discussion! The current list of types and variations is well chosen and a list type is implicitly available. The parser syntax is also in sync with this behavior.

Reply:
This has been moved to https://github.com/apache/flink/pull/21008 alongside with more tests. The list type is not sufficient for this kind of configuration. It's not a logical limitation to support many value types but not allow them in maps as values.